### PR TITLE
Bug 1245666 - Implement FTE Slideshow

### PR DESCRIPTION
**Pull request by chrafuse**

### CODE REVIEW

**Comment by mozilla-autolander-deprecated** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

File: add-on/panels/js/slideshow.jsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var loop = loop || {};
+loop.slideshow = (function(_, mozL10n) {
+  "use strict";
+
+// App state
```
Comment (Standard8) [inline add-on/panels/js/slideshow.jsx]: Please can you extract the slideshow part of this that was copied from the other repo, and put it in a file in add-on/panels/vendor. At the top of the file, please reference the source, and add their license (see urlRegExps.js for an example of where we've done this before).

In this file, we can keep the init and the data structure, and keep the mpl license.

File: add-on/panels/css/slideshow.css [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  background: none;
+}
+
+* {
+  box-sizing: border-box;
+  padding: 0;
+  margin: 0;
+  line-height: 2.4rem;  /* or 1.3rem original*/
+}
+.slideshow {
```
Comment (Standard8) [inline add-on/panels/css/slideshow.css:5]: Like I've just commented for the react code, please move this to a separate css file under add-on/panels/vendor as well, and update the license in a similar way.

Our specific css, e.g. html, body and everything under `/* slide-content */` I think can stay in this file.

File: ui/ui-showcase.jsx [hunk 3]
```diff
@@ -671,10 +671,21 @@
   });
 
   var App = React.createClass({
-
     render: function() {
       return (
         <ShowCase>
+          // Bug 1245666 - select & style or implement react slideshow component
+          <Section name="FTESlideShow">
```
Comment (Standard8) [inline ui/ui-showcase.jsx]: This doesn't work at the moment and breaks the showcase, so I suggest dropping it for now.

File: add-on/panels/css/slideshow.css [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  background-color: transparent;
+  background-repeat: no-repeat;
+  background-position: center;
+  padding: 0;
+  float: right;
+  margin-top: -20px;
+}
+.slide1 {
+  height: 100%;
+  width:  100%;
```
Comment (Standard8) [inline add-on/panels/css/slideshow.css]: `.slide` already has height & width of 100%, so I don't think these are necessary.

File: add-on/panels/css/slideshow.css [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  background: #7ec24c;
+}
+
+.slide4-image {
+  background-image: url(../../shared/img/firefox-hello_tour-slide-04.svg);
+  height: 240px;
+  width: 240px;
+  background-size: 320px 320px;
+}
```
Comment (Standard8) [inline add-on/panels/css/slideshow.css]: nit: newline at end of file please.

File: add-on/panels/js/slideshow.jsx [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+      ["GetLoopPref", "legal.privacy_url"],
+      ["GetLoopPref", "remote.autostart"],
+      ["GetUserProfile"],
+      ["GetFxAEnabled"],
+      ["GetDoNotDisturb"],
+      ["GetHasEncryptionKey"],
+      ["IsMultiProcessActive"]
+    ];
+    return loop.requestMulti.apply(null, requests.concat(prefetch)).then(function(results) {
+      console.log("in then");
```
Comment (Standard8) [inline add-on/panels/js/slideshow.jsx]: nit: please drop all the `console.log`s from this file.

File: add-on/panels/js/slideshow.jsx [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+          }),
+          text: mozL10n.get("fte_slide_4_copy", {
+            brandShortname: mozL10n.get("brandShortname")
+          })
+        }
+      ];
+      // Notify the window that we've finished initalization and initial layout
+      // var evtObject = document.createEvent("Event");
+      // evtObject.initEvent("loopPanelInitialized", true, false);
+      // window.dispatchEvent(evtObject);
```
Comment (Standard8) [inline add-on/panels/js/slideshow.jsx]: nit: please drop the comments.

File: add-on/panels/js/slideshow.jsx [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+
+      // XXX new code starts here
```
Comment (Standard8) [inline add-on/panels/js/slideshow.jsx]: I think this comment can be dropped as well.

File: add-on/panels/js/slideshow.jsx [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+      state.data = data;
+      render(state);
+
+
+      setTimeout(function() {
+        state.data = data;
+        render(state);
+      }, 1000);
```
Comment (Standard8) [inline add-on/panels/js/slideshow.jsx]: This `setTimeout` feels like a bad workaround. Is it really necessary?

File: add-on/panels/js/slideshow.jsx [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+    });
+  }
+
+  return {
+    Slideshow: Slideshow,
+    // GettingStartedView: GettingStartedView,
```
Comment (Standard8) [inline add-on/panels/js/slideshow.jsx]: nit: comment to remove.

File: add-on/panels/css/slideshow.css [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+.slide-layout {
+  padding-top: 50px;
+  padding-left: 50px;
+  padding-bottom: 20px;
+  width: 600px;
+}
+
+.slide-layout h2 {
+  font-family: "Open Sans";
+  /*font-size: 25pt;*/
```
Comment (Standard8) [inline add-on/panels/css/slideshow.css]: Please drop the commented-out line here and in `.slide-layout .slide-text`

File: add-on/panels/css/slideshow.css [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+
+/* slide content */
+.slide-layout {
+  padding-top: 50px;
+  padding-left: 50px;
+  padding-bottom: 20px;
+  width: 600px;
+}
+
+.slide-layout h2 {
```
Comment (Standard8) [inline add-on/panels/css/slideshow.css]: Please use a child selector (`>`)

File: add-on/panels/css/slideshow.css [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+  /*font-size: 25pt;*/
+  font-size: 2.8rem;
+  font-weight: 600;
+  white-space: normal;
+  color: #fff;
+  line-height: 3rem;
+  margin-bottom: 10px;
+}
+
+.slide-layout .slide-text {
```
Comment (Standard8) [inline add-on/panels/css/slideshow.css]: Please use a child selector (`>`)

File: add-on/panels/css/slideshow.css [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+  font-family: "Open Sans";
+  /*font-size: 18pt;*/
+  font-size: 2rem;
+  white-space: normal;
+  color: #fff;
+  line-height: 2.8rem;
+}
+
+.slide img {
```
Comment (Standard8) [inline add-on/panels/css/slideshow.css]: I'd suggest changing this to `.slide-layout > img` - its more consistent with the above, and allows use of the child selector.

File: add-on/panels/css/slideshow.css [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+  box-sizing: border-box;
+  border: none;
+  background-color: transparent;
+  background-repeat: no-repeat;
+  background-position: center;
+  padding: 0;
+  float: right;
+  margin-top: -20px;
+}
+.slide1 {
```
Comment (Standard8) [inline add-on/panels/css/slideshow.css:55]: nit: please insert a blank line before this.

File: add-on/panels/css/slideshow.css [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+.control-panel {
+  height: 60px;
+  background: #fff;
+  width: 100%;
+}
+
+.controls {
+}
```
Comment (Standard8) [inline add-on/panels/css/slideshow.css]: When you move this section to the vendor based css file, please also remove the empty css blocks.

File: add-on/panels/js/slideshow.jsx (hunk 1 above)
Comment (chrafuse) [inline add-on/panels/js/slideshow.jsx]: Copied the used code but could not separate cleanly.  Using the original slideshow.jsx running code the vendor version is just for license.  The slideshow had scope issues from the separate function closures used.

Comment (chrafuse) [inline add-on/panels/js/slideshow.jsx]: @Standard8 cleaned up most comments.  The license issue still needs addressing.

**Pull request merged**